File: problems/2566-maximum-difference-by-remapping-a-digit/solution.java
class Solution {
    public int minMaxDifference(int num) {
        String s = String.valueOf(num);
        int n = s.length();

        char maxDigit = ' ';
        for (int i = 0; i < n; i++) {
            if (maxDigit == ' ') {
                maxDigit = s.charAt(i);
                break;
            }
        }

        String maxNumStr = s.replace(maxDigit, '9');
        int maxNum = Integer.parseInt(maxNumStr);

        char minDigit = ' ';
        for (int i = 0; i < n; i++) {
            if (minDigit == ' ') {
                minDigit = s.charAt(i);
                break;
            }
        }

        String minNumStr = s.replace(minDigit, '0');
        int minNum = Integer.parseInt(minNumStr);
        

        return maxNum - minNum;
    }
}

File: problems/2566-maximum-difference-by-remapping-a-digit/solution.py
class Solution:
    def minMaxDifference(self, num: int) -> int:
        s = str(num)
        
        max_val = 0
        for i in range(len(s)):
            digit = s[i]
            temp = s.replace(digit, '9')
            max_val = int(temp)
            break
        
        min_val = 0
        for i in range(len(s)):
            digit = s[i]
            temp = s.replace(digit, '0')
            min_val = int(temp)
            break
            
        return max_val - min_val